src/components/Select/Select.stories.tsx

```diff
@@ -0,0 +1,223 @@
+          <>
+            <Select.Button
+              // Because we're using a render prop to completely control the styling and icon of the
+              // button, we need to configure this component to render as a Fragment. Otherwise we'd
+              // render two, nested buttons.
+              as={React.Fragment}
+            >
+              {() => (
+                <button
+                  aria-expanded={open}
+                  className={styles['function-children__button']}
+                >
+                  {selectedOption?.label || 'Select'}
+                  <Icon
+                    className={styles['function-children__icon']}
+                    name="filter-list"
+                    purpose="decorative"
+                  />
+                </button>
+              )}
+            </Select.Button>
+            <Select.Options>
+              {exampleOptions.map((option) => (
+                <Select.Option key={option.key} value={option}>
+                  {option.label}
+                </Select.Option>
+              ))}
+            </Select.Options>
+          </>
+        )}
+      </Select>
+    </div>
+  );
+}
+
+export const Default: StoryObj = {
+  render: () => (
+    <InteractiveExampleUsingChildren aria-label="Favorite Animal" />
+  ),
+};
+
+export const DefaultWithVisibleLabel: StoryObj = {
+  render: () => (
+    <InteractiveExampleUsingChildren
+      aria-label="Favorite Animal"
+      labelComponent={<Select.Label>Favorite Animal</Select.Label>}
+    />
+  ),
+};
+
+export const Compact: StoryObj = {
+  render: () => (
+    <InteractiveExampleUsingChildren
+      aria-label="Favorite Animal"
+      variant="compact"
+    />
+  ),
+};
+
+// TODO-AH: fix
+export const CompactWithOptionsRightAligned: StoryObj = {
+  render: () => (
+    <InteractiveExampleUsingChildren
+      aria-label="Favorite Animal"
+      optionsAlign="right"
+      variant="compact"
+    />
+  ),
+};
+
+// TODO-AH: fix
+export const SeparateButtonAndMenuWidth: StoryObj = {
+  render: () => (
+    <InteractiveExampleUsingChildren
+      aria-label="Favorite Animal"
+      optionsClassName={styles['separate-button-and-menu-width']}
+    />
+  ),
+};
+
+export const UsingChildrenProp: StoryObj = {
+  render: () => (
+    <InteractiveExampleUsingChildren
+      labelComponent={<Select.Label>Favorite Animal</Select.Label>}
+    />
+  ),
+};
+
+export const UsingChildrenPropAndNoVisibleLabel: StoryObj = {
+  render: () => (
+    <InteractiveExampleUsingChildren aria-label="Favorite Animal" />
+  ),
+};
+
+export const CompactUsingChildrenPropAndNoVisibleLabel: StoryObj = {
+  render: () => (
+    <InteractiveExampleUsingChildren
+      aria-label="Favorite Animal"
+      variant="compact"
+    />
+  ),
+};
+
+export const UsingFunctionChildrenProp: StoryObj = {
+  render: () => <InteractiveExampleUsingFunctionChildren />,
+};
+
+// This story just opens the dropdown automatically so chromatic can test it.
```

Comment:
1. You can use [`isChromatic`](https://www.chromatic.com/docs/ischromatic) to add play functions to the above stories if you want to snap the open state similarly (unless ofc your intention was to snap the button)
2. If you don't want to snap, you can always [disable](https://www.chromatic.com/docs/ignoring-elements#ignore-stories) the snaps in Chromatic
3. Something about timing and animations may not snap very nicely so can be flaky in Chromatic, just something to watch out for
  - Did just see a possibly new Chromatic feature regarding pausing [CSS animations](https://www.chromatic.com/docs/animations#css-animations) if you wanted to check that out

Reply:
Oh we may have to do a little bit of a dive into some of this. I will add this flag as I would like one of these to have a snap with it open. 

(this particular test was brought over from the `Dropdown` implementation)